### DISCUSSION TITLE
Error Codes

### DISCUSSION THREAD

**Discussion by ghost** [GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200#40] [General]
Not an issue, but I thought you may be interested in the 400+ error codes I added from the documentation. I only did address 3000, which is the address for the first error in the error buffer. In normal usage, this should be enough.
[errors.txt](https://github.com/user-attachments/files/17893614/errors.txt)

**Comment by GyroGearl00se** [comment]
Hi @a-fall ,

thank you for that. So far I've been skipping this on purpose, as the list is quite long :)
I'll move this to the discussion section. whoever is interested, can just add your code snippet into the froeling.yaml.

Right now I'm not able to test it, as I don't have a ESP plugged in to my froeling anymore.

I'm currently developing a Home Assistant integration (probably will be installable via HACS once ready) where you'd just need a Ethernet/Wifi -> RS485/RS232 device. (Modbus TCP)
Not entirely sure yet - But maybe I'll also have a look into reusing the ESP32 for this purpose.

**Reply by ghost** [reply]
Hi @GyroGearl00se ,

sounds very interesting! It works flawlessly for me using your ESP32 integration. I just had to swap around the two addresses 12 and 13 (I think) to get the correct fan percentage, as described in another discussion on here. This seems to vary between firmware versions apparently. This has now finally allowed me to remove internet access from my oven!

**Reply by GyroGearl00se** [reply]
I'm glad this solution works fine for you.

I think (hope) that the solution with Modbus TCP and a Home Asistant integration (via HACS) could be much easier for a lot of people. As some don't like to "mess around" :) with an ESP module.
I think I'll give it a shot. Probably first inital release here on GitHub beginning of next year.

I'm not entirely sure about the register 12 & 13. I just went with the information I've found in the Modbus Kommunikationsprotocol (pdf).
But didn't fully confirm each and every entity :)
If everyone else has the same issue - I'm happy to accept pull requests and change it accordingly.
